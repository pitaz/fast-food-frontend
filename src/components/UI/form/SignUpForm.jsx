import React from 'react';
import PropTypes from "prop-types";
import { Link, Redirect } from "react-router-dom";
import ErrorMessage from '../error/ErrorMessage.jsx';
import TextInput from '../input/TextInput.jsx';

const SignUpForm = (props) => {
  const { error, email, errors, name, password, password_confirmation, onChange, onSubmit} = props;
 return (
  <div>
  <form>
  <div className="card-header">
  {
    (error && error.message) ? (
        <p id="error-message">{error.message}</p>
    ) : ''
  }
    <p>Sign up</p>
  </div>
  <div className="card-content form-group">
    <label>
      <b>Username</b>
    </label>
    <TextInput
      type="text"
      placeholder="Enter Username"
      name="name"
      field="name"
      value={name}
      onChange={onChange}
      required
    />
    {!error || errors === undefined ? '' : <ErrorMessage errors={errors.name} />}
    <label>
      <b>Email</b>
    </label>
    <TextInput
      type="email"
      placeholder="Email"
      name="email"
      field="email"
      value={email}
      onChange={onChange}
      required
    />
    {!error || errors === undefined ? '' : <ErrorMessage errors={errors.email} />}
    <label>
      <b>Password</b>
    </label>
    <TextInput
      type="password"
      placeholder="Enter Password"
      name="password"
      field="password"
      onChange={onChange}
      value={password}
      required
    />
    {!error || errors === undefined ? '' : <ErrorMessage errors={errors.password} />}
    <label>
      <b>Confirm Password</b>
    </label>
    <TextInput
      type="password"
      placeholder="Confirm Password"
      name="password_confirmation"
      field="password_confirmation"
      onChange={onChange}
      value={password_confirmation}
      required
    />
    {!error || errors === undefined ? '' : <ErrorMessage errors={errors.password_confirmation} />}
    <button type="submit" onClick={onSubmit} >Submit </button>
    <p>
      Already Registered? <Link to="/signin">Sign In</Link>
    </p>
  </div>
</form>
  </div>
);
};

SignUpForm.propTypes = {
  name: PropTypes.string,
  email: PropTypes.string,
  password: PropTypes.string,
  password_confirmation: PropTypes.string,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
  error: PropTypes.instanceOf(Object),
  errors: PropTypes.instanceOf(Object),
};
export default SignUpForm;
