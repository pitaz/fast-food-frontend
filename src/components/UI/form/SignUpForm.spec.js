/* eslint-disable react/no-children-prop */
/* eslint-disable no-undef */
import React from 'react';
import expect from 'expect';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUpForm from './SignUpForm.jsx';


let wrapper;
let mockedState;
let props;

describe('The SignUp Component Test Suite', () => {
  beforeEach(() => {
  mockedState = {
    auth: {
      isAuthenticated: true,
      user: {
        id: 1,
        role: 'admin',
        email: 'peter@mail.com',
        iat: 1549688481,
        exp: 1549774881
      },
      error: {}
    },
  };
  props = {
    action: {
      deleteError: jest.fn(),
      signUpUser: jest.fn(),
    },
    error: {},
    auth: false,
  };
  });

  it('should match the snapshot', () => {
    wrapper = shallow(<SignUpForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render <SignUpFormForm /> Component', () => {
    wrapper = shallow(<SignUpForm {...props} />);
    expect(wrapper.length).toBe(1);
    expect(wrapper.find(SignUpForm)).toBeDefined();
  });


  // it('should test for onchange', () => { 
  //   wrapper = shallow(<SignUpForm {...props} />);

  //   const event = {
  //     target: { name: 'email', value: 'test@test.com' },
  //     preventDefault: () => { },
  //   };

  //   wrapper.instance().onChange(event);
  //   expect(wrapper.state().email).toEqual('test@test.com');
  //   expect(wrapper.instance().state.email).toEqual(event.target.value);
  // });

  
  // it('should test for onsubmit', () => { 
  //   wrapper = shallow(<SignUp {...props} />);

  //   const event = {
  //     target: { name: 'email', value: 'testtest.com' },
  //     preventDefault: () => { },
  //   };

  //   wrapper.instance().onChange(event);
  //   expect(wrapper.state().error).toEqual(undefined);
  //   expect(wrapper.instance().state.email).toEqual(event.target.value);
  // });
});
