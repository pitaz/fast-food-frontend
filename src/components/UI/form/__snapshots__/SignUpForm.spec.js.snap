// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SignUp Component Test Suite should match the snapshot 1`] = `
<div>
  <form>
    <div
      className="card-header"
    >
      <p>
        Sign up
      </p>
    </div>
    <div
      className="card-content form-group"
    >
      <label>
        <b>
          Username
        </b>
      </label>
      <TextInput
        field="name"
        name="name"
        placeholder="Enter Username"
        required={true}
        type="text"
      />
      <label>
        <b>
          Email
        </b>
      </label>
      <TextInput
        field="email"
        name="email"
        placeholder="Email"
        required={true}
        type="email"
      />
      <label>
        <b>
          Password
        </b>
      </label>
      <TextInput
        field="password"
        name="password"
        placeholder="Enter Password"
        required={true}
        type="password"
      />
      <label>
        <b>
          Confirm Password
        </b>
      </label>
      <TextInput
        field="password_confirmation"
        name="password_confirmation"
        placeholder="Confirm Password"
        required={true}
        type="password"
      />
      <button
        type="submit"
      >
        Submit 
      </button>
      <p>
        Already Registered? 
        <Link
          replace={false}
          to="/signin"
        >
          Sign In
        </Link>
      </p>
    </div>
  </form>
</div>
`;
